### PR TITLE
Ops 975 user object fix

### PR DESCRIPTION
**Pull request by xlorepdarkhelm**
## What changed

Made User object get loaded with RTK, added some tests around the created_by component to make sure it always has a value in it.

## Issue

#975 

## How to test

N/A Bug fix.

## Screenshots

N/A

## Links

N/A

### CODE REVIEW

File: frontend/src/pages/budgetLines/PreviewTable.jsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ export const PreviewTable = ({ handleDeleteBudgetLine = () => {}, readOnly = fal
         .sort((a, b) => Date.parse(a.created_on) - Date.parse(b.created_on))
         .reverse();
 
-    let loggedInUser = useSelector((state) => state.auth.activeUser.full_name);
+    let loggedInUser = useSelector((state) =>
+        state.auth?.activeUser ? state.auth.activeUser.full_name : "Not logged in"
```
Comment (fpigeonjr) [inline frontend/src/pages/budgetLines/PreviewTable.jsx:24]: Maybe we can make a reusable helper function to display the user name. 1st try `full_name`, if not available then use `email`, and last option something like `unknown user`.

**Review by fpigeonjr — CHANGES_REQUESTED** [review]
would be nice to have a helper function to display logged in user's full name, email, or unknown.

File: frontend/src/pages/budgetLines/PreviewTable.jsx [hunk 2]
```diff
@@ -11,6 +11,14 @@ import { TotalSummaryCard } from "./TotalSummaryCard";
 import { formatDate } from "../../helpers/utils";
 import "./PreviewTable.scss";
 
+export const loggedInName = (activeUser) => {
```
Comment (fpigeonjr) [inline frontend/src/pages/budgetLines/PreviewTable.jsx:14]: Would it be possible to move this to our helpers file `frontend/src/helpers/utils.js`?

**Review by fpigeonjr — CHANGES_REQUESTED** [review]
Would it be possible to move this to our helpers file `frontend/src/helpers/utils.js`?

File: frontend/src/pages/budgetLines/PreviewTable.jsx (hunk 2 above)
Comment (xlorepdarkhelm) [reply frontend/src/pages/budgetLines/PreviewTable.jsx]: Done.

File: frontend/src/components/UI/Form/CreateBudgetLinesForm.jsx [hunk 3]
```diff
@@ -171,6 +171,7 @@ export const CreateBudgetLinesForm = ({
                     </div>
                 ) : (
                     <button
+                        id="add-budget-line"
```
Comment (fpigeonjr) [inline frontend/src/components/UI/Form/CreateBudgetLinesForm.jsx:174]: learned that the `data-cy` property is the recommended way to target elements for cypress tests.  So here we could do something like `data-cy="add-budget-line-btn"`

File: frontend/src/helpers/utils.js [hunk 4]
```diff
@@ -44,3 +44,11 @@ export const convertCodeForDisplay = (list_name, code) => {
     }
     return code;
 };
+
+export const loggedInName = (activeUser) => {
```
Comment (fpigeonjr) [inline frontend/src/helpers/utils.js:48]: thank you

**Pull request merged**